backend/pkg/logger/messages/logger.go

```diff
@@ -0,0 +1,113 @@
+package messages
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/info"
+)
+
+const (
+	Name abstraction.LoggerName = "messages"
+)
+
+type Logger struct {
+	// An atomic boolean is used in order to use CompareAndSwap in the Start and Stop methods
+	running  *atomic.Bool
+	fileLock *sync.RWMutex
+	// initialTime fixes the starting time of the log
+	initialTime time.Time
+	// infoIdMap is a map that contains the file of each info packet
+	infoIdMap map[string]io.WriteCloser
```

Comment:
just as a note, the last year every info message was stored to the same file, I like that you split it by board, but if you are going to do so, use the board name instead of the id (because the id is a number that you need to cross reference with the ADE and most of the time we do not remember to keep that in mind).

I would say that you should at least keep a map from board id (number) to board name (string) (use the `abstraction.Board` types and definitions for this) which you use to get the name from the id, and then keep it the way you are doing it (one file per board) or just keep everything to the same file and just add a new field with the board to the log

Reply:
I like the order the ```BoardId```discrimination gives to the logger, and further subloggers can be added for better data distribution. I look forward to those future changes, so I'm trying to make the code as uniform and modular as possible to follow the "sublogger" philosophy along with the development.